src/plugins/workspace/public/components/workspace_form/select_data_source_panel.tsx

```diff
@@ -3,149 +3,169 @@
-      <EuiSpacer size="s" />
-      {selectedDataSources.map(({ id, title }, index) => (
-        <EuiCompressedFormRow
-          key={index}
-          isInvalid={!!errors?.[index]}
-          error={errors?.[index]?.message}
-          fullWidth
-        >
-          <EuiFlexGroup alignItems="flexEnd" gutterSize="m">
-            <EuiFlexItem style={{ maxWidth: 400 }}>
-              <EuiCompressedComboBox
-                data-test-subj="workspaceForm-select-dataSource-comboBox"
-                singleSelection
-                options={dataSourcesOptions}
-                selectedOptions={
-                  id
-                    ? [
-                        {
-                          label: title,
-                          value: id,
-                        },
-                      ]
-                    : []
-                }
-                onChange={(selectedOptions) => handleSelect(selectedOptions, index)}
-                placeholder="Select"
-              />
-            </EuiFlexItem>
-            <EuiFlexItem style={{ maxWidth: 332 }}>
-              <EuiButtonIcon
-                color="danger"
-                aria-label="Delete data source"
-                iconType="trash"
-                display="empty"
-                size="m"
-                onClick={() => handleDelete(index)}
-                isDisabled={false}
-              />
-            </EuiFlexItem>
-          </EuiFlexGroup>
-        </EuiCompressedFormRow>
-      ))}
-
-      <EuiSmallButton
-        fill
-        fullWidth={false}
-        onClick={handleAddNewOne}
-        data-test-subj={`workspaceForm-select-dataSource-addNew`}
-      >
-        {i18n.translate('workspace.form.selectDataSourcePanel.addNew', {
-          defaultMessage: 'Add New',
-        })}
-      </EuiSmallButton>
+      <EuiSpacer size="m" />
+      <EuiFlexGroup alignItems="center" gutterSize="s">
+        {isDashboardAdmin && selectedItems.length > 0 && assignedDataSources.length > 0 && (
+          <EuiFlexItem grow={false}>{removeButton}</EuiFlexItem>
+        )}
+        {isDashboardAdmin && (
+          <EuiFlexItem grow={false}>{addOpenSearchConnectionsButton}</EuiFlexItem>
+        )}
+        {isDashboardAdmin && (
+          <EuiFlexItem grow={false}>{addDirectQueryConnectionsButton}</EuiFlexItem>
+        )}
+      </EuiFlexGroup>
+      <EuiSpacer size="m" />
+      <EuiFlexItem style={{ maxWidth: 768 }}>
+        {assignedDataSources.length > 0 && renderTableContent()}
+      </EuiFlexItem>
+      {modalVisible && chrome && (
+        <AssociationDataSourceModal
+          savedObjects={savedObjects}
+          assignedConnections={assignedDataSourceConnections}
+          closeModal={() => setModalVisible(false)}
+          handleAssignDataSourceConnections={handleAssignDataSources}
+          http={http}
+          mode={toggleIdSelected as AssociationDataSourceModalMode}
+          notifications={notifications}
+          logos={chrome?.logos}
```

Comment:
Nit: you checked the existence of `chrome` already, optional chain is not required here
```suggestion
          logos={chrome.logos}
```

Reply:
thx I forgot to delete it